tensorflow/lite/micro/examples/mnist_lstm/evaluate_test.py

```diff
@@ -26,21 +26,27 @@
 PREFIX_PATH = resource_loader.get_path_to_datafile("")
 
 
-class LSTMModelTest(test_util.TensorFlowTestCase):
+class LSTMFloatModelTest(test_util.TensorFlowTestCase):
 
   model_path = os.path.join(PREFIX_PATH, "trained_lstm.tflite")
   input_shape = (1, 28, 28)
   output_shape = (1, 10)
 
   tflm_interpreter = tflm_runtime.Interpreter.from_file(model_path)
+  np.random.seed(42)  #Seed the random number generator
+
+  def testInputErrHandling(self):
+    wrong_size_image_path = os.path.join(PREFIX_PATH, "samples/resized9.png")
+    with self.assertRaisesWithPredicateMatch(ValueError,
+                                             "Invalid input image shape"):
+      evaluate.predict_image(self.tflm_interpreter, wrong_size_image_path)
 
   def testCompareWithTFLite(self):
     tflite_interpreter = tf.lite.Interpreter(model_path=self.model_path)
```

Comment:
Let's start enforcing reference kernels with TfLite when we compare the TFLM interpreter against TfLite?

@paulinesho

Reply:
Agreed, so we can remove the +/- 1 tolerance.

Reply:
@paulinesho  the +/-1 tolerance is for the python interpreter test right? Bit exactness is not pursued here. The test logic is to compare TFLite float with TFLM float first. Then TFLM quant-dequant result is compared with TFLM float.  Lastly, the prediction accuracy of the quant model. Everything is compared in floats.

Reply:
The overall testing logic should be: TF/keras (source of truth) -> TFLITE (float) -> TFLM (float) ->  TFLM (quant). Reference kernels should be used when comparing across frameworks. Correct me if I am wrong

Reply:
right, thanks for the explanation @rewu93. I still think we should be using the tflite interpreter with the reference kernels everywhere in the TFLM repo as a matter of best practice. I'm going to approve this PR and if you could file a bug and follow-up with a different PR that makes such a change across the board, that would be excellent.